fdbclient/MultiVersionTransaction.actor.cpp

```diff
@@ -989,64 +1013,119 @@ void MultiVersionApi::setCallbacksOnExternalThreads() {
 
 	callbackOnMainThread = false;
 }
-
 void MultiVersionApi::addExternalLibrary(std::string path) {
 	std::string filename = basename(path);
+	// we need at least one external library thread to run this library.
+	threadCount = std::max(threadCount, 1);
```

Comment:
Are the accesses to `threadCount` and `externalClientDescriptions` thread safe?

Reply:
threadCount is immutable once setupNetwork happens (from what I can tell this method will never be called -- we certainly don't support adding more external libraries after startup).  Later in the original versions of this function, we modify a std::map without a lock, so presumably this is OK too.

I see some places where we ensure setNetworkOption (which transitively calls this) either always runs on the main thread, or always runs while protected by a mutex.  I'm not sure what the rationale was for having multiple ways to protect the same set of variables, but I did my best to not change it.

Reply:
The earlier version of this function held a mutex. I believe we don't have any restrictions on concurrent calls to this function, so if we are modifying shared state I think it needs to be done with the mutex.

Reply:
The removal of the mutex (and the one in the other version of this method) was unintentional.  I figured it out after reading the comment below.